Python: Update change note for 1.23. (#2443)
Semmle/ql: change-notes/1.23/analysis-python.md

## General improvements

### Python 3.8 support

Python 3.8 syntax is now supported. In particular, the following constructs are parsed correctly:

- Assignment expressions using the "walrus" operator, such as `while chunk := file.read(1024): ...`.
- The positional argument separator `/`, such as in `def foo(a, /, b, *, c): ...`.
- Self-documenting expressions in f-strings, such as `f"{var=}"`.

### General query improvements

> In an effort to deprecate the `Object` API (e.g. `ClassObject`) in favour of the

Reviewer: My suggestion here and in the line below is intended to state this more positively.
```suggestion
Following the replacement of the `Object` API (for example, `ClassObject`) in favor of the
```

> `Value` API (e.g. `ClassValue`), many of the standard queries have been updated

Reviewer: ```suggestion
`Value` API (for example, `ClassValue`) in the 1.21 release, many of the standard queries have been updated
```

|----------------------------|------------------------|------------|
| Unreachable code | Fewer false positives | Analysis now accounts for uses of `contextlib.suppress` to suppress exceptions. |
| `__iter__` method returns a non-iterator | Better alert message | Alert now highlights which class is expected to be an iterator. |

> | Explicit return in __init__ method | Fewer false positives | Instances where the `__init__` method returns the value of a call to a procedure are no longer flagged. |

Reviewer: Please make the following change to this column.
```suggestion
| Explicit return in __init__ method | Fewer false positive results | Instances where the `__init__` method returns the value of a call to a procedure are no longer flagged. |
```

Author: Done.

| Non-iterable used in for loop | Fewer false positives | `__aiter__` is now recognized as an iterator method. |
| Unused import | Fewer false positives | Instances where a module is used in a forward-referenced type annotation, or only during type checking are no longer flagged. |
| Module-level cyclic import | Fewer false positives | Instances where one of the links in an import cycle is never actually executed are no longer flagged. |
| Undefined export | Fewer false positives | Instances where an exported value may be defined in a module that lacks points-to information are no longer flagged. |

> | Unreachable code | Fewer false positives | Unreachable `else` branches that do nothing but `assert` their non-reachability are no longer flagged. |

Reviewer: Please can we have these sorted alphabetically and can you add the query identifiers to the first column.

Author: Done.

## Changes to QL libraries

* Django library now recognizes positional arguments from a `django.conf.urls.url` regex (Django version 1.x)
* Instances of the `Value` class now support the `isAbsent` method, indicating

>   whether the `Value` in question is missing points-to information, but has been

Reviewer: Possibly simpler as:

```suggestion
  whether that `Value` lacks points-to information, but inference
```

>   inferred to likely exist anyway. For instance, if a file contains `import

Reviewer: ```suggestion
  suggests that it exists. For instance, if a file contains `import
```
